Exit Command 수행

flowinto10 opened the pull request.

flowinto10 committed:
resolve conflict

flowinto10 committed:
resolve conflict

Ellen99999999 committed:
[feature] SSD Shell FullRead, Full Write 기능 구현

변경점 : 신규 개발. FullRead, Full Write 기능 구현 + TDD
- fullread 는 lba 0-99 를 하나하나 읽는다. ssd_output.txt 는 매번 생성 되는 것으로 가정하고 한다.
- fullwrite 는 lba 0-99 를 동일 데이터로 써준다.

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[feature] SSD Shell FullRead, Full Write 기능 구현

변경점 : 신규 개발. FullRead, Full Write 기능 구현 + TDD
- fullread 는 lba 0-99 를 하나하나 읽는다. ssd_output.txt 는 매번 생성 되는 것으로 가정하고 한다.
- fullwrite 는 lba 0-99 를 동일 데이터로 써준다.

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[feature] SSD Shell FullRead, Full Write 기능 구현

변경점 : 신규 개발. FullRead, Full Write 기능 구현 + TDD
- fullread 는 lba 0-99 를 하나하나 읽는다. ssd_output.txt 는 매번 생성 되는 것으로 가정하고 한다.
- fullwrite 는 lba 0-99 를 동일 데이터로 써준다.

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[Feature] fullread, fullwrite 기능 구현 conflict 해결

변경점 : 충돌 해결함.

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
Merge branch 'feature/fullread_fullwrite' of https://github.com/flowinto10/CRA_TeamBest_SSDShell into feature/fullread_fullwrite

# Conflicts:
#	TeamBest_SSD_Shell/SSD_Shell.cpp

Ellen99999999 committed:
Merge branch 'feature/fullread_fullwrite' of https://github.com/flowinto10/CRA_TeamBest_SSDShell into feature/fullread_fullwrite

# Conflicts:
#	TeamBest_SSD_Shell/SSD_Shell.cpp
#	TeamBest_SSD_Shell/test_ssd_shell.cpp

flowinto10 committed:
﻿[Feature] Exit Command 처리

1. 변경점 :
- Exit command 처리
- Run 함수에 parsing 및 command 처리 추가

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

flowinto10 committed:
[Feature] Exit Command 처리

1. 변경점 :
- Exit command 처리
- Run 함수에 parsing 및 command 처리 추가

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -56,27 +53,46 @@ class SSDDriver {
 
 class SSDShell {
 public:
+	struct ParsingResult {
```
검증을 위해 ParsingResult 를 Public으로 올렸습니다.

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -56,27 +53,46 @@ class SSDDriver {
+			std::istringstream iss(line);
 
 			// 3. parsing command
+			ProcessParseInvalid(line);
```
원석님이 구현하신  Parsing 함수를 Run에서 호출하도록 하였습니다. (+ terminal 입력)

Ellen99999999 replied:
ProcessParseInvalid() 의 리턴값은 활용안하는건가요?

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -56,27 +53,46 @@ class SSDDriver {
+			}
+
 			// 4. process command
+			if (ExcuteCommand(parsingresult) == true) {
```
Parsing 결과를 입력으로 받아, Command를 수행합니다.

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -110,7 +126,40 @@ class SSDShell {
+
+		case EXIT:
+			cout << "Exit" << endl;
+			ret = true;
```
Exit Command 일때 true를 return하여 Shell program을 종료하도록 합니다.

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.h:
해당파일은 아직 include 되지 않아, 프로그램 실행과 무관합니다.
.cpp 파일의 class선언을 분리하기 위해 추가해두었습니다만... 아직 작업은 진행하지 않았습니다.

flowinto10 commented on TeamBest_SSD_Shell/test_ssdshell1.cpp:
```diff
@@ -12,3 +12,10 @@ TEST(ShellTS, CheckRunMethod) {
+	SSDShell* ssdShell = new SSDShell();
+	SSDShell::ParsingResult parsingresult{Command::EXIT, 0, " ", InvalidType::NO_ERROR};
+
+	EXPECT_EQ(true,  ssdShell->ExcuteCommand(parsingresult));
```
Exit Command 확인을 위해 TC추가하였습니다.

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -110,7 +126,40 @@ class SSDShell {
+			break;
+
+		case READ:
+			cout << "[Read] LBA " << GetAddress() << " : " << ReadSsdOutputFile(GetAddress()) << endl;
```
이 부분은 제가 만든거랑 겹치네요
ReadSsdOutputFile() 에서 "[Read] LBA " << GetAddress() << " : " <<  부분을 포함해서 return 하도록 되어 있는데
함수 본연의 역할을 하도록 ReadSsdOutputFile() 안에서는 nand 에서 읽은 데이터만 리턴하도록 업데이트 해놓는게 나을거 같네요. 그건 제가 수정해놓겠습니다.

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -110,7 +126,40 @@ class SSDShell {
+			break;
+
+		case FULL_READ:
+			// todo
```
fullwrite, fullread 는 제꺼 들어가면 여기 끼워 넣겠습니다. ㅋㅋ

Ellen99999999 reviewed (approved):
LGTM!!!!!!!!!!!!!!!!!!

Ellen99999999 committed:
Merge pull request #10 from flowinto10/feature/fullread_fullwrite

Feature/fullread fullwrite

youngchul-hwang reviewed (approved):
수고하셨습니다!

flowinto10 committed:
[Feature] Exit Command 처리

1. 변경점 :
- Exit command 처리
- Run 함수에 parsing 및 command 처리 추가

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC
